openedx/openedx-platform
# Convert Discussion Module to XBlock

Converts the Discussion XModule into an XBlock.

**JIRA tickets**: Initial work for [YONK-327](https://openedx.atlassian.net/browse/YONK-327), [OC-656](https://tasks.opencraft.com/browse/OC-656).

**Screenshots**:

The inline discussion module's appearance is unchanged.

**Sandbox URL**: 
- LMS: http://pr12582.sandbox.opencraft.com/
- Studio: http://studio-pr12582.sandbox.opencraft.com/

**Testing instructions**:

_0. Do not install this changeset yet._
1. Create a course.
2. Add a couple of units and a couple of discussions (stock modules) to it.
3. Post some posts, comments and responses and otherwise interact with discussions: follow/unfollow, upvote/downvote, etc.
4. Install this changeset.
5. Make sure LMS reruns installing dependencies; if not, at least run `pip install -r requirements/edx/local.txt` - it'll install discussion XBlock and remove discussion XModule.
6. Make sure all the threads, comments and responses are still visible, upvote/downvote count is correct, etc.
7. Check that still can post threads/comments/responses, upvote/downvote, etc.

**Reviewers**
- [x] @andy-armstrong 
- [x] @robrap
- [x] @ormsbee  

**Pre-merge**
- [x] rebase from master
